res/controllers/Traktor-Kontrol-S2-MK1-hid-scripts.js

```diff
@@ -1473,6 +1473,15 @@ const setKnobParameter = function(group, key, value, calibration) {
     }
     engine.setValue(group, key, calibratedValue);
 };
+const setKnobParameterEq = function(group, key, value, calibration) {
+    let calibratedValue;
+    if (value <= calibration.center) {
+        calibratedValue = script.absoluteLin(value, 0, 1, calibration.min, calibration.center);
+    } else {
+        calibratedValue = script.absoluteLin(value, 1, 2, calibration.center, calibration.max);
+    }
+    engine.setValue(group, key, calibratedValue);
+};
```

Comment:
I'm curious why this is required compared to the simple full-range implementation of the S4 Mk2 for example
https://github.com/mixxxdj/mixxx/blob/main/res/controllers/Traktor-Kontrol-S4-MK2-hid-scripts.js#L1489-L1491

Reply:
To center the knobs exactly. The calibration of knobs has 3 points: left, center, right.

Reply:
These 3 values are stored in the memory of the controller, after running the Native Instruments hardware calibration procedure: https://support.native-instruments.com/hc/en-us/articles/210274845-Fader-and-Knob-Calibration-of-TRAKTOR-KONTROL-Devices#Recalibrating%20your%20TRAKTOR%20KONTROL%20MK1%20/%20MK2%20devices%20including%20KONTROL%20Z1%20/%20Z2
This PR adds the usage of these calibration values.

Reply:
Okay, but why is the (parameter) range 0...1...2 ?
For [EQ paramters](https://manual.mixxx.org/2.3/en/chapters/appendix/mixxx_controls.html#control-[EffectRack1_EffectUnitN_EffectM]-parameterK) it's 0..1..4 (documented at https://manual.mixxx.org/2.3/en/chapters/appendix/mixxx_controls.html#control-[ChannelN]-filterLow

Reply:
It is not anymore. I switched to `setParameter` which is 0…1. The Eq is only 0…1…4 if the filter is not biquad.

Reply:
This needs clarification IMO. It just doesn't make any sense to me to adjust the EQ knob mapoing for a certain filter type.
@JoergAtGithub Can you share some insight maybe?

@leifhelm  Did you try `engine.setValue` and scale 0..0.5..1 ?

For reference
https://github.com/mixxxdj/mixxx/blob/2.4/src/effects/backends/builtin/equalizer_util.h#L10-L17

Reply:
I think this is already fixed in the last commit:
- engine.setValue uses the specified range from the manual
- engine.setParameter uses the normalized range from 0...1
see: https://github.com/mixxxdj/mixxx/blob/main/res/controllers/engine-api.d.ts#L45

Reply:
> Did you try engine.setValue and scale 0..0.5..1 ?

Yes. Then the eq can only be attenuated. e.g. when turning the knob all the way to the right the knob in the UI is in the center position.

This is fixed in the current version. Now all knobs point the same direction on the controller and in the UI. This is accomplished with the `setParameter` function which takes normalized input (independent of eq setting).

Reply:
yeah, I think now we got it.
sorry, I confused setValue :x:  and setParameter :heavy_check_mark: 

FYI I'm neither a js nor HID expert, so I'll not do an in-depth review of the code, I'm just skimming through and ask for clarification ; )